### DISCUSSION TITLE
Clarification on inputTypes key usage when the software supports any file type

### DISCUSSION THREAD

**Discussion by evilaliv3** [publiccodeyml/publiccode.yml#122] [Q&A]
While working on the [Globaleaks](https://github.com/globaleaks/GlobaLeaks/blob/main/publiccode.yml) publication we are experiencing difficulties giving a value to the key ```inputTypes```.

Similarly to a Content Management System (CMS), GlobaLeaks could manage every kind of file so that  the current ```inputTypes``` seems to require us to list an unlimited set of mime types with significant inefficiency.

I suggest this key could be extended / documented clarifying how this condition should be handled.

**Comment by libremente** [comment]
Thanks @evilaliv3. We should definitely take a look at that key, any input (pun intended :smile:) is appreciated!

**Comment by evilaliv3** [comment]
Thank you.

Actually i do not have useful suggestions and i think for ech key like this you need a boolean subkey like "SupportAny" to be set by default to false.

I think the final solution identified should be choosen considering similar issues like  https://github.com/publiccodeyml/publiccode.yml/issues/95

**Comment by bfabio** [comment]
Moved to GitHub Discussions

<sub>[Why?](https://github.com/publiccodeyml/publiccode.yml/discussions/104)</sub>

**Comment by bfabio** [comment]
I might be wrong, but i've always felt that `{input,output}Types` were influenced by [Desktop files](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) (`genericName` as well).

In general, that's a pretty desktop-centric key that is not really all that relevant besides for `softwareType: standalone/desktop`, and *even* for those applications brings a little value and is deemed not worth to keep updated. That's a guaranteed for incomplete information.

Case in point: not a single publiccode.yml in the Italian software catalog uses it.

IMO we should just deprecate it.

**Reply by libremente** [reply]
Totally agree @bfabio

**Comment by ruphy** [comment]
@bfabio I confirm, in the original design they were influenced by Desktop files and by the idea of helping administrations evaluate solutions which use open formats. Given the scarse usage by software vendors too, it might make sense to scrap it in a future release.
Are there thoughts from others?
